extensions/SemanticMediaWiki/src/MediaWiki/Hooks/ArticleDelete.php

```diff
@@ -19,74 +23,78 @@
 class ArticleDelete {
 
 	/**
-	 * @var Wikipage
+	 * @var WikiPage $wikiPage
 	 */
 	private $wikiPage = null;
 
 	/**
 	 * @since  2.0
 	 *
-	 * @param Wikipage $wikiPage
+	 * @param WikiPage $wikiPage
 	 */
 	public function __construct( &$wikiPage, &$user, &$reason, &$error ) {
 		$this->wikiPage = $wikiPage;
 	}
 
+	public function process() {
+		$deferredCallableUpdate =
+			ApplicationFactory::getInstance()->newDeferredCallableUpdate( function () {
+				$this->doDelete( $this->wikiPage->getTitle() );
+			} );
+
+		$deferredCallableUpdate->pushToDeferredUpdateList();
+
+		return true;
+	}
+
 	/**
-	 * @since 2.0
+	 * @since 3.0
 	 *
-	 * @return true
+	 * @param Title $title
 	 */
-	public function process() {
+	public function doDelete( Title $title ) {
 
 		$applicationFactory = ApplicationFactory::getInstance();
-		$eventHandler = EventHandler::getInstance();
 
-		$title = $this->wikiPage->getTitle();
 		$store = $applicationFactory->getStore();
+		$subject = DIWikiPage::newFromTitle( $title );
 
-		$semanticDataSerializer = $applicationFactory->newSerializerFactory()->newSemanticDataSerializer();
+		$semanticDataSerializer =
+			$applicationFactory->newSerializerFactory()->newSemanticDataSerializer();
 		$jobFactory = $applicationFactory->newJobFactory();
 
-		$deferredCallableUpdate = $applicationFactory->newDeferredCallableUpdate( function() use( $store, $title, $semanticDataSerializer, $jobFactory, $eventHandler ) {
-
-			$subject = DIWikiPage::newFromTitle( $title );
-			wfDebugLog( 'smw', 'DeferredCallableUpdate on delete for ' . $subject->getHash() );
+		// Instead of Store::getSemanticData, construct the SemanticData by
+		// attaching only the incoming properties indicating which entities
+		// carry an actual reference to this subject
+		$semanticData = new SemanticData( $subject );
 
-			$parameters['semanticData'] = $semanticDataSerializer->serialize(
-				$store->getSemanticData( $subject )
-			);
+		$properties = $store->getInProperties( $subject );
 
-			$jobFactory->newUpdateDispatcherJob( $title, $parameters )->insert();
+		foreach ( $properties as $property ) {
+			// Avoid doing $propertySubjects = $store->getPropertySubjects( $property, $subject );
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this commented out code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)